gravity/process_manager/supervisor_manager.py

```diff
@@ -243,6 +247,32 @@ def __get_supervisor(self):
         options.realize(args=["-c", self.supervisord_conf_path])
         return supervisorctl.Controller(options).get_supervisor()
 
+    def _check_path_length(self):
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        bind_path = self.supervisord_sock_path
+        try:
+            os.unlink(bind_path)
```

Comment:
It looks like this method is called any time the class is instantiated, so supervisord may be running and that would make this unsafe?

Reply:
you're right - this is a no-go.